### PR TITLE
update: Mandatory access control explainer

### PR DESCRIPTION
**Pull request by dngray**
Changes proposed in this PR:

- Closes: https://github.com/privacyguides/privacyguides.org/issues/1861
- Closes: https://github.com/privacyguides/privacyguides.org/issues/2008

The intention of this PR is to explain what mandatory access control is, and how technologies like Snapd use it to confine apps. For further direct confinement, mention [ConfinedUsers](https://fedoraproject.org/wiki/SIGs/ConfinedUsers).

<summary>

- [x] I agree to the terms listed below:
  <details><summary>Contribution terms (click to expand)</summary>
  1) I am the sole author of this work.
  2) I agree to grant Privacy Guides a perpetual, worldwide, non-exclusive, transferable, royalty-free, irrevocable license with the right to sublicense such rights through multiple tiers of sublicensees, to reproduce, modify, display, perform, relicense, and distribute my contribution as part of this project.
  3) I have disclosed any relevant conflicts of interest in my post.
  4) I agree to the Community Code of Conduct.
  </details>

### CODE REVIEW

**Review by redoomed1 — COMMENTED** [review]
Some minor suggestions, but I like the succinctness of this section

When this PR is marked as ready for review, consider replacing the "mandatory access control" link in Arch [section](https://www.privacyguides.org/en/os/linux-overview/#arch-based-distributions) of the Linux Overview. The existing Wikipedia link needs to be replaced anyway, based on Jonah's suggestion from the team chat.

---
I also suggest adding a sentence or two about which MAC Fedora and Tumbleweed use, in the same way as the last sentence in the [Microcode](https://www.privacyguides.org/en/os/linux-overview/#proprietary-firmware-microcode-updates) section and the second sentence of the [Swap](https://www.privacyguides.org/en/os/linux-overview/#swap) section.

However, I would hold off on this since Tumbleweed is considering switching to [SELinux by default](https://lists.opensuse.org/archives/list/factory@lists.opensuse.org/thread/YN4TCBCU4A2V5G2MWR5EWYF46267BO7F/) for new installs by the end of the year. Currently, Tumbleweed [defaults](https://en.opensuse.org/Portal:SELinux) to AppArmor in the installer, with the option to [choose](https://en.opensuse.org/Portal:SELinux/Setup) SELinux instead.

**Comment by dngray** [comment]
> I also suggest adding a sentence or two about which MAC Fedora and Tumbleweed use, in the same way as the last sentence in the [Microcode](https://www.privacyguides.org/en/os/linux-overview/#proprietary-firmware-microcode-updates) section and the second sentence of the [Swap](https://www.privacyguides.org/en/os/linux-overview/#swap) section.

Good idea.

> However, I would hold off on this since Tumbleweed is considering switching to [SELinux by default](https://lists.opensuse.org/archives/list/factory@lists.opensuse.org/thread/YN4TCBCU4A2V5G2MWR5EWYF46267BO7F/) for new installs by the end of the year. Currently, Tumbleweed [defaults](https://en.opensuse.org/Portal:SELinux) to AppArmor in the installer, with the option to [choose](https://en.opensuse.org/Portal:SELinux/Setup) SELinux instead.

Could just say that. I'm pretty sure [MicroOS already](https://en.opensuse.org/Portal:MicroOS/SELinux) defaults to SELinux. I can see them not wanting to manage policies for both. Likely the MCS stuff is necessary.

File: docs/os/linux-overview.md [hunk 1]
```diff
@@ -82,6 +82,10 @@ Additionally, we recommend **against** these two Arch derivatives specifically:
 
 We recommend **against** using the Linux-libre kernel, since it [removes security mitigations](https://phoronix.com/news/GNU-Linux-Libre-5.7-Released) and [suppresses kernel warnings](https://news.ycombinator.com/item?id=29674846) about vulnerable microcode.
 
+### Mandatory access control
+
+Mandatory access control is a set of additional security controls which help to confine parts of the system such as apps and system services. The two common forms of mandatory access control found in Linux distributions are [SELinux](https://github.com/SELinuxProject) and [AppArmor](https://apparmor.net). SELinux on [Fedora](https://docs.fedoraproject.org/en-US/quick-docs/selinux-getting-started) confines some Linux containers, virtual machines and service daemons by default. AppArmor is used by the snap daemon for [sandboxing](https://snapcraft.io/docs/security-sandboxing) snaps which have [strict](https://snapcraft.io/docs/snap-confinement) confinement such as [Firefox](https://snapcraft.io/firefox). There is a community effort to confine more parts of the system in Fedora with the [ConfinedUsers](https://fedoraproject.org/wiki/SIGs/ConfinedUsers) special interest group.
```
Comment (redoomed1) [inline docs/os/linux-overview.md:87]:
```suggestion
Mandatory access control is a set of additional security controls which help to confine parts of the system such as apps and system services. The two common forms of mandatory access control found in Linux distributions are [SELinux](https://github.com/SELinuxProject) and [AppArmor](https://apparmor.net). While Fedora uses SELinux by default, Tumbleweed [defaults](https://en.opensuse.org/Portal:SELinux) to AppArmor in the installer, with an option to [choose](https://en.opensuse.org/Portal:SELinux/Setup) SELinux instead.

SELinux on [Fedora](https://docs.fedoraproject.org/en-US/quick-docs/selinux-getting-started) confines some Linux containers, virtual machines and service daemons by default. AppArmor is used by the snap daemon for [sandboxing](https://snapcraft.io/docs/security-sandboxing) snaps which have [strict](https://snapcraft.io/docs/snap-confinement) confinement such as [Firefox](https://snapcraft.io/firefox). There is a community effort to confine more parts of the system in Fedora with the [ConfinedUsers](https://fedoraproject.org/wiki/SIGs/ConfinedUsers) special interest group.
```
Splitting the section into two after adding the sentence you suggested

File: docs/os/linux-overview.md [hunk 2]
```diff
@@ -82,6 +82,12 @@ Additionally, we recommend **against** these two Arch derivatives specifically:
...
 
+### Mandatory access control
+
+Mandatory access control is a set of additional security controls which help to confine parts of the system such as apps and system services. The two common forms of mandatory access control found in Linux distributions are [SELinux](https://github.com/SELinuxProject) and [AppArmor](https://apparmor.net). While Fedora uses SELinux by default, Tumbleweed [defaults](https://en.opensuse.org/Portal:SELinux) to AppArmor in the installer, with an option to [choose](https://en.opensuse.org/Portal:SELinux/Setup) SELinux instead.
+
+SELinux on [Fedora](https://docs.fedoraproject.org/en-US/quick-docs/selinux-getting-started) confines Linux containers, virtual machines and service daemons by default. AppArmor is used by the snap daemon for [sandboxing](https://snapcraft.io/docs/security-sandboxing) snaps which have [strict](https://snapcraft.io/docs/snap-confinement) confinement such as [Firefox](https://snapcraft.io/firefox). There is a community effort to confine more parts of the system in Fedora with the [ConfinedUsers](https://fedoraproject.org/wiki/SIGs/ConfinedUsers) special interest group.
```
Comment (redoomed1) [inline docs/os/linux-overview.md:89]:
```suggestion
SELinux on [Fedora](https://docs.fedoraproject.org/en-US/quick-docs/selinux-getting-started) confines Linux containers, virtual machines, and service daemons by default. AppArmor is used by the snap daemon for [sandboxing](https://snapcraft.io/docs/security-sandboxing) snaps which have [strict](https://snapcraft.io/docs/snap-confinement) confinement such as [Firefox](https://snapcraft.io/firefox). There is a community effort to confine more parts of the system in Fedora with the [ConfinedUsers](https://fedoraproject.org/wiki/SIGs/ConfinedUsers) special interest group.
```

**Review by redoomed1 — APPROVED** [review]
LGTM

**Pull request merged**